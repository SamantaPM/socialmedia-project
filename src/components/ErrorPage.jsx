import React from 'react';
import { Link, useRouteError } from 'react-router-dom';
import { HiExclamation } from 'react-icons/hi';
import logo from '../assets/logo.png';

const ErrorPage = () => {
  const error = useRouteError();
  console.error(error.error);

  return (
      <>
        <div className='w-full h-screen bg-gray-50 flex items-center justify-center '>
          <div className='flex flex-col gap-1 md:gap-3 rounded-full bg-white p-10 md:p-20 text-center items-center shadow-sm'>
          <Link to="/"><img src={logo} className='mb-7 md:mb-11 w-44 md:w-275'/></Link>
            <div className='flex flex-col justify-center items-center'>
              <HiExclamation fontSize={60} />
              <h3>Page Not Found</h3>
            </div>
            <p>Sorry, the page you are looking<br />for could not be found.</p>
            <Link to="/" className='font-semibold underline'>Go Home</Link>
          </div>
        </div>
      </>
  );
};

export default ErrorPage;
