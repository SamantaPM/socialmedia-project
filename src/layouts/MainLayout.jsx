import React, { useEffect } from 'react';
import { useNavigate, Outlet } from 'react-router-dom';
import Navbar from '../components/Navbar';
import Spinner from '../components/Spinner';
import Sidebar from '../components/Sidebar';
import SidebarMobile from '../components/SidebarMobile';
import useAuthStore from '../store/authStore';

const MainLayout = () => {
  const navigate = useNavigate();
  const { userProfile } = useAuthStore();
  useEffect(() => {
    if (!userProfile) navigate('/login');
  }, []);

  return (
    <div className="flex flex-col md:flex-row bg-gray-50">
      <div className="hidden md:flex  flex-initial h-screen">
        <Sidebar />
      </div>
      <SidebarMobile />
      <div className="flex flex-col flex-1 h-screen pb-2 overflow-y-scroll">
        <div className="px-2 md:px-5">
          <Navbar />
          <div className="h-full">
            {(navigate.state === 'loading') && <Spinner />}
            <Outlet />
          </div>
        </div>
      </div>

    </div>
  );
};

export default MainLayout;
