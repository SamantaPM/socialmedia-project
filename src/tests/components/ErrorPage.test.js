import React from 'react';
import { create } from 'react-test-renderer';
import ErrorPage from '../../components/ErrorPage';
import MainLayout from '../../layouts/MainLayout';
import { renderWithRouter } from '../utils/renderWithRouter';

console.error = jest.fn();
console.warn = jest.fn();
const routes = {
  path: '/',
  element: <MainLayout />,
  errorElement: <ErrorPage />
};

beforeEach(() => {
  console.error.mockClear();
  console.warn.mockClear();
});

describe('<ErrorPage /> tests:', () => {
  test('should render <ErrorPage /> correctly', () => {
    const tree = create(renderWithRouter(routes, [], {
      initialEntries: ['/', '/sdawdfa'],
      initialIndex: 1
    }, true)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should render console warn message', () => {
    renderWithRouter(routes, [], {
      initialEntries: ['/', '/sdawdfa'],
      initialIndex: 1
    });

    expect(console.warn).toHaveBeenCalledWith('No routes matched location \"/sdawdfa\" ');
  });
});
