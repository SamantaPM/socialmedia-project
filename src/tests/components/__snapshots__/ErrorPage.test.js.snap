// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage /> tests: should render <ErrorPage /> correctly 1`] = `
<div
  className="w-full h-screen bg-gray-50 flex items-center justify-center "
>
  <div
    className="flex flex-col gap-1 md:gap-3 rounded-full bg-white p-10 md:p-20 text-center items-center shadow-sm"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        className="mb-7 md:mb-11 w-44 md:w-275"
        src="logo.png"
      />
    </a>
    <div
      className="flex flex-col justify-center items-center"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        fontSize={60}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
          fillRule="evenodd"
        />
      </svg>
      <h3>
        Page Not Found
      </h3>
    </div>
    <p>
      Sorry, the page you are looking
      <br />
      for could not be found.
    </p>
    <a
      className="font-semibold underline"
      href="/"
      onClick={[Function]}
    >
      Go Home
    </a>
  </div>
</div>
`;
